src/aggregators/coevolve.jl

```diff
@@ -125,13 +170,78 @@ function generate_jumps!(p::CoevolveJumpAggregation, integrator, u, params, t)
 end
 
 ######################## SSA specific helper routines ########################
+function accept_next_jump!(p::CoevolveJumpAggregation, integrator, u, params, t)
+    @unpack next_jump, ma_jumps = p
+
+    num_majumps = get_num_majumps(ma_jumps)
+
+    if next_jump <= num_majumps
+        return true
+    end
+
+    @unpack cur_rates, rates, rng, urates, cur_lrates = p
+    num_cjumps = length(urates) - length(rates)
+    uidx = next_jump - num_majumps
+    lidx = uidx - num_cjumps
+
+    if lidx <= 0
+        return true
+    end
+
+    @inbounds urate = cur_rates[next_jump]
+    @inbounds lrate = cur_lrates[lidx]
+
+    s = -1
+
+    if lrate == typemax(t)
+        urate = get_urate(p, uidx, u, params, t)
+        if urate < zero(t)
+            error("urate = $(urate) < 0 for jump = $(i) at t = $(t) which is not allowed.")
+        end
+        s = urate == zero(t) ? typemax(t) : randexp(rng) / urate
+    elseif lrate < urate
+        # when the lower and upper bound are the same, then v < 1 = lrate / urate = urate / urate
+        v = rand(rng) * urate
+        # first inequality is less expensive and short-circuits the evaluation
+        if (v > lrate)
+            rate = get_rate(p, lidx, u, params, t)
+            if rate < 0
+                error("rate = $(rate) < 0 for jump = $(next_jump) at t = $(t) which is not allowed.")
+            elseif rate > urate
+                error("rate = $(rate) > urate = $(urate) for jump = $(next_jump) at t = $(t) which is not allowed.")
+            end
+            if v > rate
+                urate = get_urate(p, uidx, u, params, t)
+                s = urate == zero(t) ? typemax(t) : randexp(rng) / urate
+            end
+        end
+    elseif lrate > urate
+        error("lrate = $(lrate) > urate = $(urate) for jump = $(next_jump) at t = $(t) which is not allowed.")
+    end
+
+    if s < 0
+        return true
+    end
+
+    t = next_candidate_time!(p, u, params, t, s, lidx)
+    update!(p.pq, next_jump, t)
+    @inbounds cur_rates[next_jump] = urate
+
+    p.prev_jump = next_jump
+    generate_jumps!(p, integrator, integrator.u, integrator.p, integrator.t)
+    register_next_jump_time!(integrator, p, integrator.t)
+    u_modified!(integrator, false)
```

Comment:
@ChrisRackauckas again, just to double check. This means each time the jump condition is checked and it rejects a jump, `u_modified` is set to `false`. Is this reasonable with respect to the ODE/SDE solvers?